compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/MockDesugar.java

```diff
@@ -171,17 +176,48 @@ private BLangFunctionBody generateBody() {
     }
 
     private List<BLangStatement> generateStatements() {
-        List<BLangStatement> statements = new ArrayList<>();
-
+        // <MockFunctionObj>.functionToMock = (functionToMock);
+        BLangAssignment bLangAssignment =
+                ASTBuilderUtil.createAssignmentStmt(bLangPackage.pos, generateFieldBasedAccess(), generateRHSExpr());
         // BLangReturn Statement <retType> test:MockHandler(<MockFunctionObj>, [<args?>])
         BLangReturn blangReturn =
                 ASTBuilderUtil.createReturnStmt(bLangPackage.pos, generateTypeConversionExpression());
 
+        List<BLangStatement> statements = new ArrayList<>();
+
+        statements.add(bLangAssignment);
         statements.add(blangReturn);
 
         return statements;
     }
 
+    // <mockobj>.functionToMock =
```

Comment:
shall we make this comment more informative? may be we can add a method level comment